### PR TITLE
Make docs with make again & darkmode logo

### PR DESCRIPTION
**Pull request by clemens-fricke**
As written in the title you can make the docs again just via for example:
`make html`
`make latex`

In addition, there is a sh script that runs all examples from the examples folder (all files ending with .py).

### CODE REVIEW

File: examples/run_all_examples.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (j042) [inline examples/run_all_examples.sh:1]: Can we make this also windows compatible?

File: docs/Makefile [hunk 2]
```diff
@@ -5,16 +5,30 @@
 # from the environment for the first two.
```
[the remark this answers is not in the document]
Comment (clemens-fricke) [reply docs/Makefile:5]: The comment I think not. I have adapted the existing makefile to the new doc building pipeline

File: docs/source/conf.py [hunk 3]
```diff
@@ -49,8 +49,10 @@
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
 # html_theme = "piccolo_theme"
 html_theme = "furo"
-# html_logo = '_static/logo.png'
-# html_favicon = '_static/thumb.png'
+html_theme_options = {
```
[the remark this answers is not in the document]
Comment (clemens-fricke) [reply docs/source/conf.py:52]: They just show up in the html version of the documentation.

File: docs/source/extra_docs.py [hunk 4]
```diff
@@ -5,19 +5,17 @@
...
 """
-import os
+import pathlib
 
 import gustaf as gus
 
-this_dir = os.path.dirname(__file__)
-
 if __name__ == "__main__":
+    here = pathlib.Path(__file__).parent.resolve()
```
[the remark this answers is not in the document]
Comment (clemens-fricke) [reply docs/source/extra_docs.py:13]: This is important when rebuilding the documentation without deleting all relevant files.

File: gustaf/show.py [hunk 5]
```diff
@@ -297,9 +297,9 @@ def _vedo_showable(obj, as_dict=False, **kwargs):
...
         cmap_kwargs = obj.show_options[cmap_keys]
         # set adefault cmap if needed
-        cmap_kwargs["cname"] = obj.show_options.get("cmap", "plasma")
+        cmap_kwargs["input_cmap"] = obj.show_options.get("cmap", "plasma")
         cmap_kwargs["alpha"] = obj.show_options.get("cmapalpha", 1)
-        # add dataname
+        # add data_name
         cmap_kwargs["input_array"] = dataname
 
         # set cmap
```
[the remark this answers is not in the document]
Comment (clemens-fricke) [reply gustaf/show.py:305]: I dont understand what you want to say with this can we talk about this?

[the remark this answers is not in the document]
Comment (clemens-fricke) [reply gustaf/show.py]: change is used added

File: docs/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (clemens-fricke) [reply docs/Makefile:5]: No, it is not required to use the make file. Still, it is much easier than executing three commands after another to ensure that all files are correctly updated, especially now that we have the step where the show_options and, eventually, other features docs are generated on the fly via a Python script.

Sorry, I miss understood your original comment.

File: examples/run_all_examples_win.bat [hunk 6]
```diff
@@ -0,0 +1,24 @@
+REM Description: Run all examples and check for errors
```
Comment (j042) [inline examples/run_all_examples_win.bat:1]: haha thanks, but let's remove this then for now?

File: examples/run_all_examples.sh (hunk 1 above)
Comment (j042) [inline examples/run_all_examples.sh:1]: is there alternative to bash? does this work with zsh? perhaps python based script?

File: examples/show_gus.py [hunk 7]
```diff
@@ -1,6 +1,10 @@
 """Create gus"""
 import gustaf as gus
 
+WITH_PUPILS = False
+WITH_DARK_BACKGROUND = False
+DARK_WITH_WHITE_OVAL_BACKGROUND = False
+
```
Comment (j042) [inline examples/show_gus.py:7]: argparse ?

Comment (j042) [reply examples/show_gus.py:7]: we don't have any use with pupil now, so maybe out from this PR?

File: gustaf/show.py [hunk 8]
```diff
@@ -295,14 +306,16 @@ def _vedo_showable(obj, as_dict=False, **kwargs):
...
         cmap_kwargs["input_cmap"] = obj.show_options.get("cmap", "plasma")
         cmap_kwargs["alpha"] = obj.show_options.get("cmap_alpha", 1)
         # add data_name
         cmap_kwargs["input_array"] = data_name
 
-        # set cmap
-        vedo_obj.cmap(**cmap_kwargs)
+        # set cmap, input_cmap is removed from kwargs due to a vedo version
+        # function parameter name change. This makes it backward compatible.
+        vedo_obj.cmap(cmap_kwargs.pop("input_cmap"), **cmap_kwargs)
```
Comment (j042) [inline gustaf/show.py:318]: uff this will have merge conflict I guess

**Review by j042 — COMMENTED** [review]
Awesome, a few questions!

File: examples/run_all_examples.sh (hunk 1 above)
Comment (clemens-fricke) [reply examples/run_all_examples.sh:1]: run on my Zsh shell without a problem. I think that is more of a fallback.

**Comment by clemens-fricke** [comment]
Should we also switch to the new sphinx theme @jzwar suggested for splinepy? If so let me add it to this pr, does not really make sense to make a new doc pr in the near future again, if it can be added here.

File: examples/run_all_examples.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+"""
+import glob
+import subprocess
+import sys
+
+if __name__ == "__main__":
+    files_not_completed = []
+
+    for file in glob.glob("*.py"):
+        if file == "call_all_examples.py":
```
Comment (j042) [inline examples/run_all_examples.py:11]:
```suggestion
        if file == "run_all_examples.py":
```
maybe you meant to do this?

Comment (clemens-fricke) [reply examples/run_all_examples.py:11]: Yes , the filename was originally different and after the renaming i have forgotten to adapt this line. Nice catch

**Comment by j042** [comment]
just rebased. @clemens-fricke  could you please double check?

**Review by j042 — APPROVED** [review]
thank you !

**Pull request merged**